src/backend/escodegen/writer.wisp

```diff
@@ -1058,6 +1062,11 @@
 (install-macro! :assert expand-assert)
 
 
+(defn expand-typestr [it]
```

Comment:
Slight nitpick if let-over-lambda works, depending on how often `expand-typestr` is used it might be worth it.

Reply:
Not sure what you mean but it's basically an "unregistered" macro so the only thing that matters for the expansion is the value/expression that is passed to it.
`let` inside of it is only used for clarity (the LiveScript implementation outputs `.slice(8, -1)`, the output of `expand-typestr` here does as well). So far this code is run only in `defprotocol` so I doubt there'll be any performance issues.

Or do you mean using `let` outside of `defn`? I'm not sure how well `def` is working outside of top-level in Wisp. …Although an attempt to use `def`/`defn` inside a `let`, `defn` or even `do` seems to produce a compile error anyway (`Unknown expression type: VariableDeclaration`).

Reply:
> (the LiveScript implementation outputs `.slice(8, -1)`

Right.

> So far this code is run only in `defprotocol` so I doubt there'll be any performance issues.

Ok, seems good then.



> Or do you mean using `let` outside of `defn`? I'm not sure how well `def` is working outside of top-level in Wisp. …Although an attempt to use `def`/`defn` inside a `let`, `defn` or even `do` seems to produce a compile error anyway (`Unknown expression type: VariableDeclaration`).

Ok, I feel like that could be slightly interesting to have, but I'm not volunteering myself (at least not now), so that'll be that for now.